temporalio/client.py

```diff
@@ -188,10 +198,80 @@ def data_converter(self) -> temporalio.converter.DataConverter:
         """Data converter used by this client."""
         return self._config["data_converter"]
 
+    @overload
+    async def start_workflow(
+        self,
+        workflow: Callable[[WorkflowClass], Awaitable[WorkflowReturnType]],
+        *,
+        id: str,
+        task_queue: str,
+        execution_timeout: Optional[timedelta] = None,
+        run_timeout: Optional[timedelta] = None,
+        task_timeout: Optional[timedelta] = None,
+        id_reuse_policy: temporalio.common.WorkflowIDReusePolicy = temporalio.common.WorkflowIDReusePolicy.ALLOW_DUPLICATE,
+        retry_policy: Optional[temporalio.common.RetryPolicy] = None,
+        cron_schedule: str = "",
+        memo: Optional[Mapping[str, Any]] = None,
+        search_attributes: Optional[Mapping[str, Any]] = None,
+        header: Optional[Mapping[str, Any]] = None,
+        start_signal: Optional[str] = None,
+        start_signal_args: Iterable[Any] = [],
+    ) -> WorkflowHandle[WorkflowClass, WorkflowReturnType]:
+        ...
+
+    @overload
+    async def start_workflow(
+        self,
+        workflow: Callable[
+            [WorkflowClass, LocalParamType], Awaitable[WorkflowReturnType]
+        ],
+        arg: LocalParamType,
+        *,
+        id: str,
+        task_queue: str,
+        execution_timeout: Optional[timedelta] = None,
+        run_timeout: Optional[timedelta] = None,
+        task_timeout: Optional[timedelta] = None,
+        id_reuse_policy: temporalio.common.WorkflowIDReusePolicy = temporalio.common.WorkflowIDReusePolicy.ALLOW_DUPLICATE,
+        retry_policy: Optional[temporalio.common.RetryPolicy] = None,
+        cron_schedule: str = "",
+        memo: Optional[Mapping[str, Any]] = None,
+        search_attributes: Optional[Mapping[str, Any]] = None,
```

Comment:
Might as well make "raw" search_attributes `Mapping[str, List[str | int | bool | datetime]` (I think float is valid too).
Can be in the next PR.

Reply:
Yeah, I need to tackle search attributes. I plan on typing them better in that case. I also may wait until TS search attribute API decisions get sussed out as they apply here too.